packages/core/src/indexing/types.ts

```diff
@@ -0,0 +1,66 @@
+import type { StreamID } from '@ceramicnetwork/streamid'
+import type { DID } from 'dids'
+
+export interface IndexStreamArgs {
+  streamID: StreamID
+  model: StreamID
+  controller: DID
+  lastAnchor: Date | null
+  content: Record<string, any>
+  relationFields?: Map<string, StreamID | DID>
+}
+
+export enum Ordering {
+  CHRONOLOGICAL = 'chronological', // last_anchored_at DESC NULLS FIRST, created_at DESC
+  INSERTION = 'insertion', // created_at DESC = when an entry was added to the index
+}
+
+export interface BaseQuery {
+  model: StreamID | string
+  account?: string
```

Comment:
`DID | string`?

Reply:
No, a DID identifier is always a string. the DID class contains functionality for actions using a DID.